### DISCUSSION TITLE
Filemanager not rendering images

### DISCUSSION THREAD

**Discussion by michal-cecko** [oneduo/nova-file-manager#102] [Q&A] [answered]
Hi there, I am having a problem with filemanager, it wont render images... When i upload image, manager saves it but cant render.
<img width="360" alt="image" src="https://user-images.githubusercontent.com/70513191/202448136-a82c9d0a-137f-49b3-86e7-ac40eb42c5da.png">
Nova file manager config is set to defaults. Symlink between storage and public folder is created too. Any ideas what could be wrong?

**Comment by mikaelpopowicz** [comment]
Hi @Elvinca 

Did you configure your `APP_URL` envar ?

May you provide your `filesystems.php` configuration file ?

May you try this url in your browser ? `http{s}://${APP_URL}/storage/appJablotron.png`

**Reply by michal-cecko** [reply]
Hi there @mikaelpopowicz and thanks for your quick response.

Here is my filesystems.php config:

<img width="772" alt="image" src="https://user-images.githubusercontent.com/70513191/202698337-a9a18afa-8684-4e22-884e-b2bc5896246a.png">

My .env APP_URL is "localhost:8000" and when i try to open up the image path in browser as you suggested, it loads correctly, it just won't load in the filemanager.

**Comment by mikaelpopowicz** [comment, marked as answer]
May you provide the response of the following request `http://localhost:8000/nova-vendor/nova-file-manager/?path=%2F&perPage=15&disk=public&fieldMode=false` ?

**Reply by michal-cecko** [reply]
Here is the response JSON:

{"disk":"public","breadcrumbs":[],"folders":[{"id":"2e58e0c1ea673cd22bd3a55a2fc1177945495215","path":"\/photos","name":"photos"},{"id":"3ffbfd19c97c5654f993d3aea267c0f000d2761d","path":"\/pouzivatelia","name":"pouzivatelia"},{"id":"6466be6b32bcb114743e3d00770e08ed6938e7bf","path":"\/vseobecne","name":"vseobecne"}],"files":[{"id":"b8c35a5be58104ef965170406694a2779547716a","disk":"public","name":"appJablotron.png","path":"appJablotron.png","size":"107.07 KB","extension":"png","mime":"image\/png","url":"localhost:8000\/storage\/appJablotron.png","lastModifiedAt":"2 days ago","type":"image","exists":true,"meta":{"type":"image","width":1280,"height":895,"aspectRatio":null}}],"pagination":{"current_page":1,"last_page":1,"from":1,"to":1,"total":1,"links":[{"url":null,"label":"&laquo; Previous","active":false},{"url":"\/?page=1","label":"1","active":true},{"url":null,"label":"Next &raquo;","active":false}]}}

**Reply by mikaelpopowicz** [reply]
May you try with the following `APP_URL` http://localhost:8000 ?

**Reply by michal-cecko** [reply]
That json was already with `APP_URL` http://localhost:8000/.

**Reply by mikaelpopowicz** [reply]
> ,"url":"localhost:8000/storage/appJablotron.png"

The **http://** is not present in the above url, you may add it to your `APP_URL`.

**Reply by michal-cecko** [reply]
Hey @mikaelpopowicz, I really wouldn't say that it helps, but it really does... Somebody give me back 6hrs of my life :) Thank you, my issue is now resolved :)

**Comment by Rezrazi** [comment]
Hi @Elvinca 

As it stands we are unable to replicate your issue, and we need more debug information to be able to help you.
There are multiple ways you can do that, either set up a repo in which you replicate this issue and share it so we can try to figure out what went wrong.
Or you can create [gists](https://gist.github.com/) of your browser console log, and the failed requests in your Network tab under your browser's devtools

And meanwhile, as you have supposedly already configured your app storage, what happens when you try to reach an asset with its url ? For instance if you have a file named `image.png` in `storage/public/image.png` and your app url is `http://localhost:8000` what happens when you go to this url in your browser `http://localhost:8000/storage/image.png`

If the image does not render, the issue is then unrelated to the File Manager and you should double check your setup.

If the image does indeed render in the browser and not the file manager, we're going to need to have logs of your browser's console and Network tab.

All the best,
Charaf
